lib/cloud_controller/upload_buildpack.rb

```diff
@@ -47,8 +50,31 @@ def upload_buildpack(buildpack, bits_file_path, new_filename)
       true
     end
 
+    def extract_stack_from_buildpack(bits_file_path)
+      bits_file_path = bits_file_path.path if bits_file_path.respond_to?(:path)
+      output, _, status = Open3.capture3('unzip', '-p', bits_file_path, 'manifest.yml')
```

Comment:
Since the second output is ignored, you could [use `capture2` instead](https://ruby-doc.org/stdlib-1.9.3/libdoc/open3/rdoc/Open3.html#method-c-capture2)

- @selzoc && maryam

Reply:
certainly

Reply:
If possible, we would prefer to use the `rubyzip` gem instead of shelling out.

Reply:
Cool, we will refactor to use rubyzip instead.